docs/data/data-grid/column-definition/column-definition.md

```diff
@@ -131,6 +131,40 @@ You can check the [styling cells](/x/react-data-grid/style/#styling-cells) secti
 Cell content should not be in the tab sequence except if cell is focused.
 You can check the [tab sequence](/x/react-data-grid/accessibility/#tab-sequence) section for more information.
 
+### Using hooks inside a renderer
+
+The `renderCell` property is a function returning a React node, not a React component.
+
+If you can to use React hooks inside your renderer, you should wrap them inside a component.
+
+```tsx
+// ❌ Not valid
+const column = {
+  // ...other properties,
+  renderCell: () => {
+    const [count, setCount] = React.useState(0);
+
+    return (
+      <Button onClick={() => setCount((prev) => prev + 1)}>{count} click(s)</Button>
+    );
+  },
+};
+
+// ✅ Valid
+const CountButton = () => {
+  const [count, setCount] = React.useState(0);
+
+  return (
+    <Button onClick={() => setCount((prev) => prev + 1)}>{count} click(s)</Button>
+  );
+};
+
+const column = {
+  // ...other properties,
+  renderCell: () => <CountButton />,
+};
+```
+
```

Comment:
For instance
Or even uglier:

```tsx
apiRef.current.setState(state => ({
  ...state,
  myCustomState: { ... }
}));
apiRef.current.forceUpdate();
```

Reply:
Yes, and I feel like your message could encourage them to do it :laughing:

Reply:
Is the following less misleading?
 
> If you want to persist cell information, you should either save it either in the data grid parent or in the [row](https://mui.com/x/react-data-grid/rows/#the-updaterows-method).
Updating the `row` will rerender the row and so call `renderCell` with updated params.

Reply:
Yes
Storing in the row will often be a bad solution because it retriggers filtering / sorting / aggregation ...
We could probably have a concept of row cache at some point (something that is only accessible during rendering but persistent between mounts.
Or by working on https://github.com/mui/mui-x/issues/4160 we could make those updates lighter (but we will still update the aggregation if we follow AG Grid approach so not great to replace an internal state)